### ISSUE TITLE
Enhance internal links in notes

### ISSUE THREAD

**Issue by a2line** [geneweb/geneweb#17]
TLDR: add possibility to make internal links to individual with Sosa number instead of names.

Current documented syntax to make a link to an individual from a note is only `[[first_name/name/occ/text]]` that writes “Text” with a HTML link to the individual (link is red if ind don't exist in db). Here are other syntax not documented at the moment (thanks to Henri for this), when occurence is ommited, it's 0 by default:
1. `[[first_name/name]]` writes “First_name Name”
2. `[[first_name/NAME]]` writes “First_name NAME”
3. `[[first_name/name/text]]` writes “text” too
4.  `[[first_name/name//text]]` (3rd part empty) writes “text”, same as 3 

For records, I add that:
1. `[[first_name/NAme]]` writes a red link “First_name NAme”
2. `[[fn/n/]]` or `[[fn/n/occ/]]` or `[[fn/n//]]` (3rd/4th part empty or both empty) writes nothing, there's no link at all.

First, Henri noted that some cases should be improved because if an occurrence number is needed, [[first_name/name/1]] writes “1” (that can be a red link if no homonym carrying occ 0) and we need to write the full [[first_name/name/1/First_name Name]], but we would like it to work like 1st/2nd cases that are shorter to write. So if third part is an occurrence number (#) and there is no fourth part, we want `[[first_name/name/#]] to write “First name Name” (and correct link to ind. occ #) and not “#” (with link to occurrence 0 of that ind.). And if someone really wants the text part to be “1”, he will need to write the full `[[first_name/name/occ/1]]`.

All this said, what I wanted to ask for is a syntax to make a link to a Sosa number of an individual. On the contrary of actual syntax with full first names/names, it would have the advantage to be really short to write and to stay effective even if the target is renamed or its occurence number changed. The disadvantage is that it will only work for Sosa ancestor of current individual. Here are the syntaxes those would do the job:
1. `[[#Sosa_number]]` write `<a href="#">[[n° Sosa]] sosa_number Sosa's_Name</a>`
2. `[[#Sosa_number/text]]` :  `<a href="link_to_sosa #">text</a>` (suggested by Mehdi)

More ideas with an example. Marie Victorine LEGRAND is Sosa 9 of Marcel DUPONT. On Marcel's note I can have a link to his great-grandmother with:
1. `[[#9]]` writes link “Sosa 9 Marie Victorine LEGRAND”  (default, using case of name stored in db)
2. `[[#9/Marie Legrand]]` writes “Sosa 9 Marie Legrand” (personnalised text)
3. `[[#9u]]` “Sosa 9 Marie Victorine LEGRAND” (force uppercase, here same as default)
4. `[[#9l]]`  "Sosa 9 Marie Victorine Legrand” (force lowercase)
5. `[[#-9]]` to remove the Sosa number, “Marie Victorine LEGRAND”

Possible combinations of previous cases:
1. `[[#-9/Legrand]]` “Legrand”
1. `[[#-9l]]` “Marie Victorine Legrand”
2. `[[#-9/arrière-grand-mère Legrand]]` “arrière-grand-mère Legrand”
